File: leetcode/922.cpp
#include <iostream>
#include <queue>
#include <vector>

using namespace std;

class Solution {
   public:
    vector<int> sortArrayByParityII(vector<int>& A) {
        vector<int> even, odd;
        for (int i : A) {
            if (i % 2 == 0) {
                even.push_back(i);
            } else {
                odd.push_back(i);
            }
        }
        vector<int>::iterator even_it = even.begin();
        vector<int>::iterator odd_it = odd.begin();
        vector<int> result;

        int N = A.size();
        for (int i = 0; i < N; ++i) {
            if (i % 2 == 0) {
                result.push_back(*even_it);
                ++even_it;
            } else {
                result.push_back(*odd_it);
                ++odd_it;
            }
        }
    }
};

int main(void) {
    Solution sol;
    vector<int> nums = {4, 2, 5, 7};
    sol.sortArrayByParityII(nums);
    return 0;
}

File: leetcode/922.py
import queue


class Solution:
    def sortArrayByParityII(self, nums):
        even = list()
        odd = list()
        answer = list()
        for i in range(0, len(nums)):
            if nums[i] % 2 == 0:
                even.append(nums[i])
            else:
                odd.append(nums[i])
        total = len(even) + len(odd)

        for i in range(0, total):
            if i % 2 == 0:
                answer.append(even.pop())
            else:
                answer.append(odd.pop())
        return answer


ns = [4, 2, 5, 7]
sol = Solution()
print(sol.sortArrayByParityII(ns))
